### PR TITLE
[WIP] cmake toolchains cleanup and fix

### PR DESCRIPTION
**Pull request by dagar**
This is necessary so that cmake configure time compile tests (eg endianness) work properly. 

Needed by https://github.com/PX4/Firmware/pull/9651.

### CODE REVIEW

**Comment by dagar** [comment]
Once our own hacks are removed from the CMake toolchains things like using clang to compile nuttx will work.

File: cmake/toolchains/Toolchain-arm-linux-gnueabihf.cmake [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
 	endif()
 endforeach()
 
-FIND_PROGRAM(STRIP_TOOL "arm-linux-gnueabihf-strip")
-
-set(CMAKE_EXE_LINKER_FLAGS "-Wl,-gc-sections")
-#set(CMAKE_C_FLAGS ${C_FLAGS})
-#set(CMAKE_CXX_LINKER_FLAGS ${C_FLAGS})
-
+set(CMAKE_FIND_ROOT_PATH get_file_component(${CMAKE_C_COMPILER} PATH))
```
Comment (TSC21) [inline cmake/toolchains/Toolchain-arm-linux-gnueabihf.cmake]: I think the difference on endianness working would actually be related to this.

File: cmake/toolchains/Toolchain-arm-none-eabi.cmake [hunk 2]
```diff
@@ -1,101 +1,48 @@
...
 
-# optional compiler tools
-foreach(tool gdb gdbtui)
-	string(TOUPPER ${tool} TOOL)
-	find_program(${TOOL} arm-none-eabi-${tool})
-	if(NOT ${TOOL})
-		#message(STATUS "could not find ${tool}")
-	endif()
-endforeach()
+set(CMAKE_FIND_ROOT_PATH get_file_component(${CMAKE_C_COMPILER} PATH))
```
Comment (TSC21) [inline cmake/toolchains/Toolchain-arm-none-eabi.cmake]: same thing here.

**Comment by dagar** [comment]
It's not done yet.

**Comment by TSC21** [comment]
Yeah CI seems to have symlink complaints.

**Comment by TSC21** [comment]
@dagar any progress on this?

**Comment by TSC21** [comment]
Still trying to figure out why does the endianness check results on having the `CMAKE_SYSTEM_PROCESSOR` becoming invalid. @dagar do we probably need to force check the endianness on the system we are cross compiling to before the `TestBigEndian` Cmake test runs?

**Comment by dagar** [comment]
No, I think we just need to be more careful with setting cmake variables so that the test compile stages of configure work.

**Comment by dagar** [comment]
Closer, but not quite working fully.

**Comment by TSC21** [comment]
@dagar anything that I can do to bring this further? Though the with the endianness check off, I think it is beneficial, not only to micro-CDR, but system wise, to have a clean Cmake build structure.

**Pull request closed without merging**